sns_ik_lib/utilities/sns_linear_solver.hpp

```diff
@@ -0,0 +1,113 @@
+/** @file sns_linear_solver.hpp
+ *
+ * @brief Linear Solver, used by the SNS-IK velocity IK solvers.
+ * @author: Matthew Kelly
+ *
+ * This class is used to solve linear systems. It is a wrapper for two different internal solvers:
+ *   --> psuedo-inverse solver: included for legacy support on Eigen 3.2.0
+ *   --> direct linear solver: prefered solver when available. Requries Eigen 3.3.4
+ *
+ *    Copyright 2018 Rethink Robotics
```

Comment:
Again, copyright first, separate the description with different braces

Reply:
updated!